crates/bevy_render/src/view/mod.rs

```diff
@@ -74,15 +75,37 @@ pub struct Msaa {
     /// Note that WGPU currently only supports 1 or 4 samples.
     /// Ultimately we plan on supporting whatever is natively supported on a given device.
     /// Check out this issue for more info: <https://github.com/gfx-rs/wgpu/issues/1832>
-    pub samples: u32,
+    pub level: MultiSampleLevel,
+}
+
+impl Msaa {
+    #[inline]
+    pub fn samples(&self) -> u32 {
+        self.level as u32
+    }
 }
 
 impl Default for Msaa {
     fn default() -> Self {
-        Self { samples: 4 }
+        Self {
+            level: MultiSampleLevel::Sample4,
+        }
     }
 }
 
+impl From<MultiSampleLevel> for Msaa {
+    fn from(level: MultiSampleLevel) -> Self {
+        Self { level }
+    }
+}
+
+#[derive(Clone, Copy, Reflect, PartialEq)]
+#[non_exhaustive]
+pub enum MultiSampleLevel {
+    Off = 1,
+    Sample4 = 4,
+}
```

Comment:
Shouldn't we also add `Sample2` and `Sample8` for native?

Reply:
Never mind, the support for this has been merged into Wgpu but not released yet.